src/Http/Routing/src/RouteEndpointBuilder.cs

```diff
@@ -79,9 +80,46 @@ public override Endpoint Build()
             requestDelegate,
             RoutePattern,
             Order,
-            new EndpointMetadataCollection(Metadata),
+            CreateMetadataCollection(Metadata),
             DisplayName);
 
         return routeEndpoint;
     }
+
+    private static EndpointMetadataCollection CreateMetadataCollection(IList<object> metadata)
+    {
+        if (metadata is { Count: > 0 })
```

Comment:
Is this the right place to do this?

Reply:
I was debating this too. The other option is RouteEndpointDataSource. We need to make sure that this isn't redoing what MVC is doing already too. Maybe we can consolidate the MVC logic here though. And it mightfix other EndpointDataSources for free. I'm leaning towards leaving it hear and removing the special logic in MVC.

Reply:
I don't know much about routing yet but seems a good idea to add it here and have other datasources have it fixed for free, Is there any other place where we could have it as common behavior?

Reply:
Interesting, let me do some tests with MVC to see how it behave. The `CorsApplicationModelProvider` could not be remove because it does more than that but for sure we could remove this part.

Reply:
> Interesting, let me do some tests with MVC to see how it behave. The `CorsApplicationModelProvider` could not be remove because it does more than that but for sure we could remove this part.

I like it because the `CorsApplicationModelProvider` implementation does not work for something like this:

``` c#
// Configure the HTTP request pipeline.
app.MapControllers().RequireCors("t");
```